lupa/_lupa.pyx

```diff
@@ -1667,13 +1678,20 @@ cdef _LuaTable py_to_lua_table(LuaRuntime runtime, lua_State* L, items):
     old_top = lua.lua_gettop(L)
     lua.lua_newtable(L)
     # FIXME: how to check for failure?
-
+    if recursive and mapped_objs is None:
+        mapped_objs = {}
     try:
         for obj in items:
+            if recursive:
+                if id(obj) not in mapped_objs:
+                    mapped_objs[id(obj)] = lua.lua_gettop(L)
+                else:
+                    idx = mapped_objs[id(obj)]
+                    return new_lua_table(runtime, L, <int>idx)
             if isinstance(obj, dict):
                 for key, value in (<dict>obj).items():
-                    py_to_lua(runtime, L, key, wrap_none=True)
-                    py_to_lua(runtime, L, value)
+                    py_to_lua(runtime, L, key, True, recursive, mapped_objs)
+                    py_to_lua(runtime, L, value, False, recursive, mapped_objs)
```

Comment:
As I suggested before, please use keyword arguments instead of bare `True`/`False` values.

Reply:
Ok, I've changed them all.